### ISSUE TITLE
possibly-used-before-assignment for function that is checked for call ability before calling

### ISSUE THREAD

**Issue by nikita-leontiev** [pylint-dev/pylint#9880]
### Bug description

```python
Below code leads to possibly-used-before-assignment error for function that is checked for call ability before calling:

import sys
from module import get_module_name

module_name = get_module_name()
if module_name == "module1":
    from module import func
elif module_name == "module2":
    pass
else:
    assert False

if __name__ == "__main__":
    if callable(getattr(sys.modules[__name__], "func", None)):
        func()
```

### Configuration

```ini
Without pylintrc.
```

### Command used

```shell
pylint -sn --disable=E0401,C0114 test.py
```

### Pylint output

```python
test.py:14:8: E0606: Possibly using variable 'func' before assignment (possibly-used-before-assignment)
```

### Expected behavior

No E0606 error.

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.8.10 (default, Jul 29 2024, 17:02:10)
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by jacobtylerwalls** [comment]
The error is emitted because if `get_module_name()` returns `"module2"` then the program will raise NameError.

**Comment by nikita-leontiev** [comment]
If get_module_name() returns "module2", then getattr() returns None and callable() returns False for None param.
Where can NameError occur here?
